File: fix.py
import sys

with open('src/App.tsx', 'r') as f:
    content = f.read()

import re

# Find the location of: isBooked ? "bg-red-100...
# and replace everything up to <div className="flex gap-4">

content = re.sub(r'(isBooked \? "bg-red-100[^"]+" : "bg-emerald-50[^"]+"\s*\)\s*\}\s*>).*?(<div className="flex gap-4">)', r'\1\n                                <span>{d.getDate()}</span>\n                              </div>\n                            );\n                          })}\n                        </div>\n                      </div>\n                      \2', content, flags=re.DOTALL)

with open('src/App.tsx', 'w') as f:
    f.write(content)

File: fix.js
const fs = require('fs');
let content = fs.readFileSync('src/App.tsx', 'utf8');

const target = `                                <span>{d.getDate()}</span>
                              </div>
                            );
                          })}
                        </div>
                      </div>
                      <div className="flex gap-4">`;

const current = `                               </div>                      })}
                              </div>
                            </div>
                          );
                        })()}
                      </div>
                      <div className="flex gap-4">`;

content = content.replace(current, target);
fs.writeFileSync('src/App.tsx', content);
